api/v1alpha3/kubeadm_control_plane_types.go

```diff
@@ -0,0 +1,164 @@
+	// This is a pointer to distinguish between explicit zero and not specified.
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Version defines the desired Kubernetes version.
+	// +kubebuilder:validation:MinLength:=1
+	Version string `json:"version"`
+
+	// InfrastructureTemplate is a required reference to a custom resource
+	// offered by an infrastructure provider.
+	InfrastructureTemplate corev1.ObjectReference `json:"infrastructureTemplate"`
+
+	// KubeadmConfigSpec is a KubeadmConfigSpec
+	// to use for initializing and joining machines to the control plane.
+	KubeadmConfigSpec cabpkv1.KubeadmConfigSpec `json:"kubeadmConfigSpec"`
+}
+
+// KubeadmControlPlaneStatus defines the observed state of KubeadmControlPlane.
+type KubeadmControlPlaneStatus struct {
+	// Selector is the label selector in string format to avoid introspection
+	// by clients, and is used to provide the CRD-based integration for the
+	// scale subresource and additional integrations for things like kubectl
+	// describe.. The string will be in the same format as the query-param syntax.
+	// More info about label selectors: http://kubernetes.io/docs/user-guide/labels#label-selectors
+	// +optional
+	Selector string `json:"selector,omitempty"`
+
+	// Total number of non-terminated machines targeted by this control plane
+	// (their labels match the selector).
+	// +optional
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// Total number of non-terminated machines targeted by this control plane
+	// that have the desired template spec.
+	// +optional
+	UpdatedReplicas int32 `json:"updatedReplicas,omitempty"`
+
+	// Total number of fully running and ready control plane machines.
+	// +optional
+	ReadyReplicas int32 `json:"readyReplicas,omitempty"`
+
+	// Total number of unavailable machines targeted by this control plane.
+	// This is the total number of machines that are still required for
+	// the deployment to have 100% available capacity. They may either
+	// be machines that are running but not yet ready or machines
+	// that still have not been created.
+	// +optional
+	UnavailableReplicas int32 `json:"unavailableReplicas,omitempty"`
+
+	// Initialized denotes whether or not the control plane has the
+	// uploaded kubeadm-config configmap.
+	// +optional
+	Initialized bool `json:"initialized,omitempty"`
+
+	// Ready denotes that the KubeadmControlPlane API Server is ready to
+	// receive requests.
+	// +optional
+	Ready bool `json:"ready,omitempty"`
+
+	// ErrorReason indicates that there is a problem reconciling the
+	// state, and will be set to a token value suitable for
+	// programmatic interpretation.
+	// +optional
+	ErrorReason KubeadmControlPlaneStatusError `json:"errorReason,omitempty"`
+
+	// ErrorMessage indicates that there is a problem reconciling the
+	// state, and will be set to a descriptive error message.
+	// +optional
+	ErrorMessage *string `json:"errorMessage,omitempty"`
```

Comment:
Same as above -  Based on https://github.com/kubernetes-sigs/cluster-api/issues/1670 -  I think this should maybe be renamed to `FailureMessage`?

Reply:
Addressed